ksql-engine/src/test/java/io/confluent/ksql/KsqlEngineTest.java

```diff
@@ -506,12 +506,14 @@ public void shouldNotDeleteSchemaNorTopicForTable() throws Exception {
   }
 
   @Test
-  public void shouldCleanUpInternalTopicSchemasFromSchemaRegistry() throws Exception {
+  public void shouldCleanUpInternalTopicSchemasOnTerminate() throws Exception {
     // Given:
     ksqlEngine.execute(
         "create stream s1  with (value_format = 'avro') as select * from test1;"
         + "create table t1 as select col1, count(*) from s1 group by col1;",
-        KSQL_CONFIG, Collections.emptyMap());
+        KSQL_CONFIG, Collections.emptyMap())
+        .forEach(QueryMetadata::start);
```

Comment:
It may well do. I don't think its mocked. However, Kafka is mocked.

The query needs to be started, otherwise it won't perform certain actions on `close()`.